### PR TITLE
github/workflows: move cli e2e test to separate workflow, run on windows as well

### LINKED ISSUE #318: Separate GitHub workflow for cli e2e test
Thinking we can save some time by only running the CLI e2e test when `packages/cli` is changed. If we want e2e test for the app/plugins, that should be a separate thing.

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes #318

### CODE REVIEW

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    paths: ['.github/workflows/cli.yml', 'packages/cli/**']
+    types: [opened, reopened, edited, synchronize]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (freben) [inline .github/workflows/cli.yml]: thorough, nice

**Review by soapraj — APPROVED** [review]

File: .github/workflows/cli.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        node-version: [12.x]
+
+    name: Node ${{ matrix.node-version }} on ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: find location of yarn cache
+        id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
```
Comment (freben) [inline .github/workflows/cli.yml]: is this like a hack to achieve general variable setting?

**Review by freben — APPROVED** [review]

**Comment by Rugvip** [comment]
Needs #357

**Comment by Rugvip** [comment]
Skipping the windows build for now, getting errors that I can't reproduce locally on windows

**Pull request merged**